alphastats/gui/utils/import_helper.py

```diff
@@ -233,143 +188,141 @@ def empty_session_state():
             )
 
 
-def select_columns_for_loaders(software, software_df: None):
+def show_loader_columns_selection(
+    software: str, softwarefile_df: Optional[pd.DataFrame] = None
+) -> Tuple[str, str]:
     """
     select intensity and index column depending on software
     will be saved in session state
     """
     st.write("\n\n")
-    st.markdown("### 2. Select columns used for further analysis.")
-    st.markdown("Select intensity columns for further analysis")
+    st.markdown("##### 2. Select columns used for analysis")
+    st.markdown("Select intensity columns for analysis")
 
     if software != "Other":
-        st.selectbox(
+        intensity_column = st.selectbox(
             "Intensity Column",
             options=SOFTWARE_OPTIONS.get(software).get("intensity_column"),
-            key="intensity_column",
         )
 
-        st.markdown("Select index column (with ProteinGroups) for further analysis")
+        st.markdown("Select index column (with ProteinGroups) for analysis")
 
-        st.selectbox(
+        index_column = st.selectbox(
             "Index Column",
             options=SOFTWARE_OPTIONS.get(software).get("index_column"),
-            key="index_column",
         )
 
     else:
-        st.multiselect(
+        intensity_column = st.multiselect(
             "Intensity Columns",
-            options=software_df.columns.to_list(),
-            key="intensity_column",
-        )
+            options=softwarefile_df.columns.to_list(),
+        )  # TODO why is this a multiselect?
 
         st.markdown("Select index column (with ProteinGroups) for further analysis")
 
-        st.selectbox(
+        index_column = st.selectbox(
             "Index Column",
-            options=software_df.columns.to_list(),
-            key="index_column",
+            options=softwarefile_df.columns.to_list(),
         )
 
+    return intensity_column, index_column
+
 
-def select_sample_column_metadata(df, software):
-    samples_proteomics_data = get_sample_names_from_software_file()
-    valid_sample_columns = []
+def show_select_sample_column_for_metadata(
+    df: pd.DataFrame, software: str, loader: BaseLoader
+) -> str:
+    """Show the 'select sample column for metadata' component and return the value."""
+    samples_proteomics_data = get_sample_names_from_software_file(loader)
 
-    for col in df.columns.to_list():
-        if bool(set(samples_proteomics_data) & set(df[col].to_list())):
-            valid_sample_columns.append(col)
+    valid_sample_columns = [
+        col
+        for col in df.columns.to_list()
+        if bool(set(samples_proteomics_data) & set(df[col].to_list()))
+    ]
 
     if len(valid_sample_columns) == 0:
-        st.error(
+        raise ValueError(
             f"Metadata does not match Proteomics data."
             f"Information for the samples: {samples_proteomics_data} is required."
         )
 
+    # TODO I get an ERROR: "described in Please upload proteinGroups.txt"
```

Comment:
This is unrelated to the software and can just say 'described in the column names of the uploaded proteomic data file.